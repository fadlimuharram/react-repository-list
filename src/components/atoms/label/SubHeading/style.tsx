import styled from 'styled-components';

export const Wrapper = styled.h1`
	font-family: Montserrat, sans-serif;
	font-style: normal;
	font-weight: normal;
	font-size: 1.125rem;
	line-height: 26px;
`;
