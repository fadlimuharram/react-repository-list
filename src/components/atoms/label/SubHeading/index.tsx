import React from 'react';
import { Wrapper } from './style';

const SubHeading: React.FC = ({ children }) => {
	return <Wrapper data-testid="sub-heading">{children}</Wrapper>;
};

export default SubHeading;
