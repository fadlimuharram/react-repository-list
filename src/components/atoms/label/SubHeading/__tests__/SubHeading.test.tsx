import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SubHeading from '..';

afterEach(() => {
	cleanup();
});

test('should render Hello from Sub Heading', () => {
	render(<SubHeading>Hello</SubHeading>);
	const headingElm = screen.getByTestId('sub-heading');
	expect(headingElm).toBeInTheDocument();
	expect(headingElm).toHaveTextContent('Hello');
});

test('Match Sub Heading Snapshot', () => {
	const tree = renderer.create(<SubHeading>Hello</SubHeading>).toJSON();

	expect(tree).toMatchSnapshot();
});
