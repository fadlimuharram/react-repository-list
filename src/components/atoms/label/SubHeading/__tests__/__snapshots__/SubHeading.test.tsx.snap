// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Sub Heading Snapshot 1`] = `
<h1
  className="sc-bdnxRM blitjC"
  data-testid="sub-heading"
>
  Hello
</h1>
`;
